Feat/register page

kiwidebruyne opened the pull request:
## 태스크 URL
[회원가입 페이지](https://www.notion.so/wafflestudio/f9e357a8436c4cd7a92139fe7140365f?pvs=4)


## 체크리스트
- [ ] pre-commit 성공 여부
- [x] yarn build 성공 여부
- [x] Type label 추가


## 설명
![image](https://github.com/wafflestudio/ggzz-client/assets/105846051/23784666-3145-4815-a9d7-1dfec9e370ac)

회원가입 페이지입니다. 로그인 페이지와 마찬가지로 일단 임의로 디자인해서 구현했습니다. 이메일은 일단 형식상 받기만 했습니다.

## 테스트 방법 (Optional)
`yarn start`

kiwidebruyne committed:
FIX : globals.css 수정사항 반영 및
src/pages/login/index.tsx, index.module.scss class명 변경

kiwidebruyne committed:
FIX : email 입력칸 추가 및
pages/register 페이지 login 페이지와 비슷하게 디자인 변경

kiwidebruyne committed:
FIX : email 유효성 검사에 '.' 포함여부 추가

kiwidebruyne committed:
FIX : caution 메시지 style 수정

GoGiants1 committed:
Feat: install firebase for social login

GoGiants1 committed:
feat: add google login btn

GoGiants1 committed:
Feat: init firebase app

GoGiants1 committed:
Feat: add google btn in login and sign up page

GoGiants1 commented on src/lib/firebase.ts:
```diff
@@ -0,0 +1,38 @@
+// Your web app's Firebase configuration
+// For Firebase JS SDK v7.20.0 and later, measurementId is optional
+const firebaseConfig = {
+  apiKey: process.env.REACT_APP_FIREBASE_API_KEY,
+  authDomain: process.env.REACT_APP_FIREBASE_AUTH_DOMAIN,
+  projectId: process.env.REACT_APP_FIREBASE_PROJECT_ID,
+  storageBucket: process.env.REACT_APP_FIREBASE_STORAGE_BUCKET,
+  messagingSenderId: process.env.REACT_APP_FIREBASE_MESSAGING_SENDER_ID,
+  appId: process.env.REACT_APP_FIREBASE_APP_ID,
+  measurementId: process.env.REACT_APP_FIREBASE_MEASUREMENT_ID
+};
```
@wafflestudio/ggzz-frontend 여러분들 firebase 대시보드에서 해당 정보들 받아오셔서, `.env`파일에 저장해주셔야합니다!

GoGiants1 commented:
## 회원가입
<img width="360" alt="image" src="https://github.com/wafflestudio/ggzz-client/assets/69342392/9d8c9097-1bea-4d29-a955-138bde3c4733">

## 로그인
<img width="363" alt="image" src="https://github.com/wafflestudio/ggzz-client/assets/69342392/6505cee1-5388-425d-a329-2d17c1e25c59">

GoGiants1 committed:
feat: register user info after google login

GoGiants1 committed:
refactor: registration page
- Remove email field
- Rename ID to username for consistency with api parameter name
- Create MyInfo store

GoGiants1 committed:
feat: create axios instance for interceptor logic

GoGiants1 committed:
feat: replace axios with custom axios instance

GoGiants1 committed:
fix: check user login state with session storage

GoGiants1 committed:
chore: add TODO comment
